Zend/zend_enum.c

```diff
@@ -16,11 +16,13 @@
    +----------------------------------------------------------------------+
 */
 
+#define GENERATE_ZEND_ENUM_REGISTRATIONS
```

Comment:
What's this for?

Reply:
Avoiding messages in every file including the `.h` about unused static functions (the registration functions) while compiling - this #if enables them.

Reply:
Oh, you're including the zend_enum_arginfo.h from the header. Why do you do that? The arginfo.h files are not supposed to be included in headers.

Reply:
To have the arginfo for tryfrom/from/cases available - every enum needs them.

Reply:
Also see `ZEND_ENUM_ME_END` and `ZEND_ENUM_BACKED_ME_END`

Reply:
Actually, I tried to expose them via an extern first - but if I do that via initializer assignment then the compiler complains that I cannot use non-compile time constants in initializers...

Reply:
Would it be possible to add the methods in the registration function rather than requiring them to be explicitly listed? Same as userland that is.

Reply:
I would have to malloc() some array so that the builtin_functions pointer would be adequately populated ... and then free() it ... when the class entry is destroyed. I feel quite unsure about that.
This is definitely more obvious in userland.

Reply:
Oh, you probably mean directly adding it to the function_table when you say same as userland. that probably could work out...

Reply:
Actually, I just could call zend_register_functions() - was way easier than I expected.